File: TI1.4 RESTFUL_DOTNET_SINBDD_GR06/ConUni_RestFul_Dotnet_GR06/ConUni_RestFul_Dotnet_GR06/Servicios/ConversorLongitud.cs
namespace ConUni_RestFul_Dotnet_GR06.Servicios
{
    /// <summary>
    /// Servicio de conversión de longitudes
    /// </summary>
    public class ConversorLongitud
    {
        // Metro <-> Kilómetro
        public double MetroAKilometro(double metros)
        {
            return metros / 1000.0;
        }

        public double KilometroAMetro(double kilometros)
        {
            return kilometros * 1000.0;
        }

        // Metro <-> Milla
        public double MetroAMilla(double metros)
        {
            return metros / 1609.344;
        }

        public double MillaAMetro(double millas)
        {
            return millas * 1609.344;
        }

        // Kilómetro <-> Milla
        public double KilometroAMilla(double kilometros)
        {
            return MetroAMilla(KilometroAMetro(kilometros));
        }

        public double MillaAKilometro(double millas)
        {
            return MetroAKilometro(MillaAMetro(millas));
        }
    }
}

File: Deberes_PP_Arquitectura-entregable/TI1.1 SOAP_JAVA_SINBDD_GR06/01 SERVIDOR/ConUni_Soap_Java_GR06/src/java/ec/edu/monster/services/ConversorLongitud.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.monster.services;

/**
 *
 * @author josue
 */
public class ConversorLongitud {
    
    // Metro <-> Kilómetro
    public double metroAKilometro(double metros) {
        return metros / 1000.0;
    }
    
    public double kilometroAMetro(double kilometros) {
        return kilometros * 1000.0;
    }
    
    // Metro <-> Milla
    public double metroAMilla(double metros) {
        return metros / 1609.344;
    }
    
    public double millaAMetro(double millas) {
        return millas * 1609.344;
    }
    
    // Kilómetro <-> Milla
    public double kilometroAMilla(double kilometros) {
        return metroAMilla(kilometroAMetro(kilometros));
    }
    
    public double millaAKilometro(double millas) {
        return metroAKilometro(millaAMetro(millas));
    }
}
